### DISCUSSION TITLE
Combine Zod-schemas with SWR hooks?

### DISCUSSION THREAD

**Discussion by johanneskvamme** [orval-labs/orval#1349] [Q&A] [answered]
Hello!

Is there an easy way to combine both `zod` and `SWR` with Orval?
I would like to have my data parsed by zod before it is returned by SWR, so that SWR sets error to true if the data is not in the correct format.

**Comment by melloware** [comment, marked as answer]
I think your question is similar to this one: https://github.com/anymaniax/orval/issues/839

That person is asking about Tanstack Query but I am guessing SWR is the same?

**Reply by johanneskvamme** [reply]
Oh, woops! Didn't find that one, but you are correct! The need is the same.

**Comment by gegorov** [comment]
suggested issue does not have any solution. Does any one know any way how to inject zod validation inside query fn?
